rfc/rfc-87/rfc-87.md

```diff
@@ -24,9 +24,329 @@
+   * for both serialized and deserialized representations of record key.
+   */
+  @Override
+  public int compare(RowData a, RowData b) {
+    return comparator.compare(
+        a.getString(RECORD_KEY_POS).toString(),
+        b.getString(RECORD_KEY_POS).toString());
+  }
+
+  @Override
+  public void swapKey(MemorySegment segI, int offsetI, MemorySegment segJ, int offsetJ) {
+    byte[] temp = new byte[NUM_KEY_BYTES];
+    segI.swapBytes(temp, segJ, offsetI, offsetJ, NUM_KEY_BYTES);
+  }
+
+  @Override
+  public int getNumKeyBytes() {
+    return NUM_KEY_BYTES;
+  }
+
+  @Override
+  public boolean isKeyFullyDetermines() {
+    return false;
+  }
+
+  @Override
+  public boolean invertKey() {
+    return false;
+  }
+}
+```
+
+Comparing record using _hoodie_record_key has a problem, for ComplexKeyGenerator _hoodie_record_key starts with column name.
+It makes BinaryInMemorySortBuffer usage inefficient, because BinaryInMemorySortBuffer builds index with first 8 bytes from key to make sorting faster (avoid records serialization/deserialization). 
+Here is possible to use something like FastRecordKeyComparator to trim column name from key values.
+
+```Java
+/**
+ * Comparator for record keys used in sorting operations before writing and during data compaction.
+ * Utilizes {@link SignedBytes#lexicographicalComparator()} for high-performance lexicographical comparison.
+ *
+ * <p>Writer uses this class for sorting data before writing, while the Compaction service uses it
+ * for determining the order of records.
+ */
+public class FastRecordKeyComparator implements Comparator<String> {
+  Comparator<byte[]> comparator = SignedBytes.lexicographicalComparator();
+
+  /**
+   * Compares two string representing full record keys.
+   *
+   * @return Negative if a < b, positive if a > b, 0 if equal.
+   */
+  @Override
+  public int compare(String recordKeyA, String recordKeyB) {
+    return compareBytesWithoutColumnName(
+        getRecordKeyBytes(recordKeyA),
+        getRecordKeyBytes(recordKeyB));
+  }
+
+  /**
+   * Compares two byte arrays representing record keys without column names and delimiters.
+   * 
+   * <p>These byte arrays should be a result of {@link #getRecordKeyBytes}
+   *
+   * @return Negative if a < b, positive if a > b, 0 if equal.
+   */
+  public int compareBytesWithoutColumnName(byte[] a, byte[] b) {
+    return comparator.compare(a, b);
+  }
+
+  /**
+   * Converts a record key into a byte array by extracting only values (without column name)
+   * and encoding them in UTF-8.
+   *
+   * @return Byte array representing record key without column names and delimiters, these byte arrays
+   * should only be compared using {@link #compareBytesWithoutColumnName(byte[], byte[])} method.  
+   */
+  public byte[] getRecordKeyBytes(String recordKey) {
+    return String.join("", KeyGenUtils.extractRecordKeys(recordKey))
+        .getBytes(StandardCharsets.UTF_8);
+  }
+}
+```
+
+### FlinkAppendHandle / FlinkCreateHandle
```

Comment:
What about other handles? e.g., `FlinkMergeHandle`, `FlinkMergeAndReplaceHandle`.. There should also be works for record merging based on `RowData`.